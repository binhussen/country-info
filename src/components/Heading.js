/** @format */

import { useLocation } from 'react-router-dom';

const Heading = () => {
  const location = useLocation();
  const { svg } = location?.state?.flag ?? '';
  return (
    <section className="text-white bg-slate-400 h-40 p-5 center">
      <img
        src={svg
          ?? 'https://cdn.shortpixel.ai/spai/q_lossy+w_810+to_webp+ret_img/https://www.geographyrealm.com/wp-content/uploads/2019/08/Antipodes_globe.png'}
        alt={svg}
        className="h-32 black mx-auto "
      />
    </section>
  );
};
export default Heading;
