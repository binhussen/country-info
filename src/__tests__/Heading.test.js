/** @format */

import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Heading from '../components/Heading';

describe('Heading Component', () => {
  test('Builds the snapshot of the Heading component', () => {
    const heading = render(
      <BrowserRouter>
        <Heading />
      </BrowserRouter>,
    );
    expect(heading).toMatchSnapshot();
  });
});
