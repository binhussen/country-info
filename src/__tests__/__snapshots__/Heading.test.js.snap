// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component Builds the snapshot of the Heading component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="text-white bg-slate-400 h-40 p-5 center"
      >
        <img
          class="h-32 black mx-auto "
          src="https://cdn.shortpixel.ai/spai/q_lossy+w_810+to_webp+ret_img/https://www.geographyrealm.com/wp-content/uploads/2019/08/Antipodes_globe.png"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="text-white bg-slate-400 h-40 p-5 center"
    >
      <img
        class="h-32 black mx-auto "
        src="https://cdn.shortpixel.ai/spai/q_lossy+w_810+to_webp+ret_img/https://www.geographyrealm.com/wp-content/uploads/2019/08/Antipodes_globe.png"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Heading Component snapshot of the Heading component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="text-white bg-slate-400 h-40 p-5 center"
      >
        <img
          class="h-32 black mx-auto "
          src="https://cdn.shortpixel.ai/spai/q_lossy+w_810+to_webp+ret_img/https://www.geographyrealm.com/wp-content/uploads/2019/08/Antipodes_globe.png"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="text-white bg-slate-400 h-40 p-5 center"
    >
      <img
        class="h-32 black mx-auto "
        src="https://cdn.shortpixel.ai/spai/q_lossy+w_810+to_webp+ret_img/https://www.geographyrealm.com/wp-content/uploads/2019/08/Antipodes_globe.png"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
